### PR TITLE
[receiver/k8scluster] Use newer v2 HorizontalPodAutoscaler for Kubern…

### PR DESCRIPTION
**Pull request by jvoravong**
**Description:** <Describe what has changed.>
With these changes the k8scluster receiver will be able to monitor the deprecated v2beta2 and v2 versions of the HorizontalPodAutoscaler. We need these changes for v2 HPA to properly support Kubernetes 1.26. I left in the v2beta2 support to keep support for other clusters.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/20480

**Testing:** 
Tested manually with Kubernetes 1.25. Added a unit test as well.

### CODE REVIEW

File: .chloggen/fix-k8scluster-use-v2-horizontalpodautoscaler.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: receiver/k8scluste
```
Comment (rmfitzpatrick) [inline .chloggen/fix-k8scluster-use-v2-horizontalpodautoscaler.yaml:5]:
```suggestion
component: receiver/k8scluster
```

File: receiver/k8sclusterreceiver/internal/hpa/hpa.go [hunk 2]
```diff
@@ -91,7 +92,54 @@ func GetMetrics(hpa *autoscalingv2beta2.HorizontalPodAutoscaler) []*agentmetrics
...
+
+	return []*agentmetricspb.ExportMetricsServiceRequest{
+		{
+			Resource: getResourceForHPABeta(hpa),
+			Metrics:  metrics,
+		},
+	}
+}
+
+func getResourceForHPA(hpa *autoscalingv2.HorizontalPodAutoscaler) *resourcepb.Resource {
```
Comment (rmfitzpatrick) [inline receiver/k8sclusterreceiver/internal/hpa/hpa.go:131]: Can the existing be updated to take an `ObjectMeta` so no repeated definition is required?

Comment (jvoravong) [reply receiver/k8sclusterreceiver/internal/hpa/hpa.go]: good call, done

**Review by rmfitzpatrick — APPROVED** [review]

**Review by jpkrohling — APPROVED** [review]

**Comment by jpkrohling** [comment]
Waiting on @dmitryax as the code owner.

**Review by dmitryax — APPROVED** [review]
LGTM

**Pull request merged**